app/src/main/java/org/openhdp/hdt/ui/history/HistoryFragment.kt

```diff
@@ -95,20 +97,29 @@ class HistoryFragment : Fragment() {
                 requireActivity().showText("No timestamps so far")
             }
             is HistoryViewState.StopwatchesResult -> {
-                binding.timerSelector.text = state.stopwatch.name
-                groupieSection.update(state.items.map {
-                    renderer.invoke(it)
-                })
+                renderStopwatchesResult(state)
             }
             is HistoryViewState.Error -> {
                 requireActivity().showText("Error ${state.throwable}")
             }
-            is HistoryViewState.HistoryResult -> {
-                binding.timerSelector.text = "History"
-                groupieSection.update(state.items.map {
-                    renderer.invoke(it)
-                })
-            }
+        }
+    }
+
+    private fun renderStopwatchesResult(
+        state: HistoryViewState.StopwatchesResult
+    ) {
+        binding.timerSelector.text = state.stopwatch?.name ?: "History"
+        groupieSection.update(state.items.map {
+            renderer.invoke(it)
+        })
+        state.tiemstampToEdit?.let { entry ->
+            AlertDialog.Builder(requireContext())
+                .setTitle(entry.stopwatchName)
+                .setMessage("Are you sure to delete this timestamp?")
+                .setPositiveButton("YES") { dialog, which ->
+                    viewModel.onDelete(entry)
+                }.setNegativeButton("NO") { _, _ -> }
```

Comment:
I think that something with the negative button is wrong.

Reply:
there was failing test, that's why circle-ci failed

`.setNegativeButton("NO") { _, _ -> }` works the same as
`.setNegativeButton("NO") { dialog, which -> }`